Library/Homebrew/test/os/mac/keg_spec.rb

```diff
@@ -5,28 +5,65 @@
 
   subject { described_class.new(keg_path) }
 
-  describe "#mach_o_files" do
-    let(:keg_path) { HOMEBREW_CELLAR/"a/1.0" }
+  let(:keg_path) { HOMEBREW_CELLAR/"a/1.0" }
+  let(:file) { keg_path/"lib/i386.dylib" }
 
-    before(:each) { (keg_path/"lib").mkpath }
-
-    after(:each) { subject.unlink }
+  before(:each) do
+    (keg_path/"lib").mkpath
+    cp dylib_path("i386"), file
+  end
 
-    it "skips hardlinks" do
-      cp dylib_path("i386"), keg_path/"lib/i386.dylib"
-      ln keg_path/"lib/i386.dylib", keg_path/"lib/i386_hardlink.dylib"
+  after(:each) { subject.unlink }
 
+  describe "#change_dylib_id" do
+    before(:each) do
       subject.link
-      expect(subject.mach_o_files.count).to eq(1)
     end
 
-    it "isn't confused by symlinks" do
-      cp dylib_path("i386"), keg_path/"lib/i386.dylib"
-      ln keg_path/"lib/i386.dylib", keg_path/"lib/i386_hardlink.dylib"
-      ln_s keg_path/"lib/i386.dylib", keg_path/"lib/i386_symlink.dylib"
+    it "does nothing if given id is same as file's dylib id" do
+      id = file.dylib_id
+      subject.change_dylib_id(id, file)
+      expect(file.dylib_id).to eq(id)
+    end
+
+    it "changes file's dylib id to given id" do
+      id = "blah.B.dylib"
+      file.ensure_writable do
+        subject.change_dylib_id(id, file)
+      end
+      expect(file.dylib_id).to eq(id)
```

Comment:
I'm not sure why this is failing -- `change_dylib_id` is a pretty thin wrapper over `MachO::Tools.change_dylib_id`, and it works correctly for me in isolation:

```ruby
>> m.dylib_id #=> "libfoo.1.dylib"
>> MachO::Tools.change_dylib_id('i386.dylib', 'blah.B.dylib') #=> 4164
>> m = MachO.open('i386.dylib')
>> m.dylib_id #=> "blah.B.dylib"
```

Is it possible that `file` is being overwritten somewhere before this test?

Reply:
The file object that is created when the file is opened (https://github.com/Homebrew/brew/blob/master/Library/Homebrew/vendor/macho/macho/tools.rb#L22) at has it's dylib_id changed while filename's dylib_id isn't changed.

Reply:
It looks like the test's passing now - is it flaky? Did it get resolved since?

Reply:
@mansimarkaur You're right, but that shouldn't matter unless there's a race condition somewhere (i.e., the check is being performed before `write!` occurs inside of `change_install_name`). 

I'll do some digging when I have access to my OS X machine (tomorrow morning).

Reply:
Even after the `write!`, file's dylib_id is changed while that is not true for filename.